### ISSUE TITLE
The plugin slows down the IDE a lot

### ISSUE THREAD

**Issue by thimpat** [hsz/idea-gitignore#525]
### Prerequisites

* [x] Plugin is in the latest version
* [x] Issue was not reported yet
* [ ] Stack trace (if provided) contains `mobi.hsz.idea.gitignore` package name

### Description

Right click on a directory or a file makes Intellij Idea slows down monumentally to the point that the IDE becomes unusable.

```
Stack trace if available.
```

### Steps to Reproduce

1. Right click on a file
2. Add to .gitignore

**Expected behavior:** 
It should not slow down the IDE

**Actual behavior:** 
It kills the IDE responsiveness

**Reproduces how often:** 
All the time

### Versions
2.4.0

**Plugin:** 

```
Can be found in Settings > Plugins > .ignore
```

**IDE:** 
Intellij IDEA + JRE 1.8.0
Ultimate Edition: 2017.3.5
```
Can be found in Help > About. Click on the copy icon on the left.
```

**OS:** 
Windows 10
```
Information about operation system - type, version.
```

### Additional Information

_Any additional information, configuration or data that might be necessary to reproduce the issue._

**Comment by dmariaa** [comment]
I had to disable the plugin to be able to work, it definitely slows down the IDE.

**Comment by addshore** [comment]
I also had to disable the plugin.

I filed a report with jetbrains before they told me that it was this plugin.

https://youtrack.jetbrains.com/issue/WI-39562

The ticket includes CPU snapshots.

**Comment by hsz** [comment]
@addshore Thank you for sharing the CPU Snapshot. It was very helpful. Thanks to that I've noticed that the issue was with obtaining the VCS Root for the analyzed files:

![image](https://user-images.githubusercontent.com/108333/37869218-1a2d0fcc-2fb4-11e8-9d2f-aba828de643e.png)

I've refactored this part to use cached VCS Roots.

In addition, I've also fixed #521 _MatcherUtil.cache takes too much memory_ - it should also give us some performance benefits.
Both fixes will be released next week with `v2.5.0`.
